### PR TITLE
HyperLogLog support

### LINKED ISSUE #491: HyperLogLog data type support [JIRA: CLIENTS-979]
See basho/riak-java-client#669

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #491 (CLIENTS-979)

### CODE REVIEW

File: riak/codecs/pbuf.py [hunk 1]
```diff
@@ -605,17 +608,26 @@ def decode_map_value(self, entries):
...
+                    'Map may not contain datatype: {}'
+                    .format(dtype))
             out[(name, dtype)] = value
         return out
 
     def decode_set_value(self, set_value):
         return [bytes_to_str(string[:]) for string in set_value]
 
+    def decode_hll_value(self, hll_value):
+        return int(hll_value)
```
Comment (lukebakken) [inline riak/codecs/pbuf.py]: Note: in Python 2, `int()` thankfully does the right thing if `hll_value` requires a `long` internally.

File: riak/tests/test_datatypes.py [hunk 2]
```diff
@@ -97,19 +97,29 @@ def op(self, dtype):
     def check_op_output(self, op):
         self.assertIn('adds', op)
         self.assertItemsEqual(op['adds'], ['bar', 'foo'])
-        self.assertIn('removes', op)
```
Comment (alexmoore) [inline riak/tests/test_datatypes.py]: Did you mean to remove these lines from the Set Unit tests?

File: riak/codecs/pbuf.py [hunk 3]
```diff
@@ -662,6 +678,10 @@ def encode_map_update(self, dtype, msg, op):
                 msg.flag_op = riak.pb.riak_dt_pb2.MapUpdate.ENABLE
             else:
                 msg.flag_op = riak.pb.riak_dt_pb2.MapUpdate.DISABLE
+        else:
```
Comment (alexmoore) [inline riak/codecs/pbuf.py]: Good catch

File: riak/tests/test_datatypes.py (hunk 2 above)
Comment (lukebakken) [inline riak/tests/test_datatypes.py]: Nope!

**Comment by alexmoore** [comment]
👍  :shipit:

**Pull request merged**